### PR TITLE
Feat #115 - Decouple columns & limit from table view

### PR DESCRIPTION
**Pull request by Sm00g15**

### CODE REVIEW

File: src/reducers/app/thunks.ts [hunk 1]
```diff
@@ -157,12 +157,12 @@ export const submitQuery = () => async (dispatch, state) => {
   });
   // query = setLimit(query, limit);
   // Add this to set ordering
+  alert(attributeNames);
   query = addOrdering(
     query,
-    attributeNames.includes('block_level') ? 'block_level' : 'level',
+    !attributeNames.includes('level') ? 'block_level' : 'level',
```
Comment (anonymoussprocket) [inline src/reducers/app/thunks.ts]: These names need to be normalized at the Conseil service level

**Comment by anonymoussprocket** [comment]
Add a function similar to `getInitialColumns` called `getInitialFilters`. For example, default operations filter should be `kind = 'transaction'`

**Comment by anonymoussprocket** [comment]
Change column sorting on the selector drop down:
- visible columns in display order
- the rest in alphabetical order

**Comment by anonymoussprocket** [comment]
Format numbers with thousands separator.

**Comment by umurb** [comment]
> Changing column selection does not update or refresh view

That is the purpose of this ticket.

File: src/reducers/app/thunks.ts (hunk 1 above)
Comment (umurb) [reply src/reducers/app/thunks.ts]: Agreed. We will go over operators and attributes in another ticket

**Comment by Sm00g15** [comment]
**NOTE** default results to 5000 and results per page to 50

**Comment by Sm00g15** [comment]
I've updated this PR so the Limit button and it's associated functions have been removed, the default query limit is 5000 and the user by default will see 50 rows per page.

**Comment by developer0623** [comment]
chain

**Comment by Sm00g15** [comment]
@developer0623 that's right, b/c it's an attribute with low cardinality the only operators that's relevant to it is "Equals". @umurb this will be something to address in future tickets, low cardinality attributes should probably only see equals.

File: src/containers/CustomTable/index.tsx [hunk 2]
```diff
@@ -126,15 +121,15 @@ class CustomTable extends React.Component<Props, State> {
                 );
               })}
               {emptyRows > 0 && (
-                <TableRow style={{ height: 48 * emptyRows }}>
+                <TableRow style={{ height: 48 / emptyRows }}>
```
Comment (developer0623) [inline src/containers/CustomTable/index.tsx]: Why is it necessary?

File: src/reducers/app/thunks.ts [hunk 3]
```diff
@@ -47,10 +47,33 @@ const getAttributeNames = (attributes) => {
...
-    return newColumns;
-  } else {
+  const sorted = columns.sort((a, b) =>
+    a.displayName.localeCompare(b.displayName)
+  );
+
+  if (entity === 'operations') {
+    let operationColumns: string[] = [];
+
+    operationColumns.push(sorted.filter(c => c.name === 'timestamp')[0]);
```
Comment (developer0623) [inline src/reducers/app/thunks.ts]: Please use an filter

File: src/reducers/app/thunks.ts [hunk 4]
```diff
@@ -47,10 +47,33 @@ const getAttributeNames = (attributes) => {
...
+    operationColumns.push(sorted.filter(c => c.name === 'source')[0]);
+    operationColumns.push(sorted.filter(c => c.name === 'destination')[0]);
+    operationColumns.push(sorted.filter(c => c.name === 'amount')[0]);
+    operationColumns.push(sorted.filter(c => c.name === 'kind')[0]);
+
+    return operationColumns;
+  } else if (entity === 'accounts') {
+    let accountColumns: string[] = [];
+
+    accountColumns.push(sorted.filter(c => c.name === 'account_id')[0]);
```
Comment (developer0623) [inline src/reducers/app/thunks.ts]: Please use a filter.

File: src/containers/CustomTable/index.tsx (hunk 2 above)
Comment (Sm00g15) [reply src/containers/CustomTable/index.tsx]: It removes the whitespace below the results in the table if there's less than 50 results. An example query to try with it is: Level Equals 1 on Blocks

Comment (developer0623) [reply src/containers/CustomTable/index.tsx]: Please remove the `{emptyRows >0 && }`

File: src/reducers/app/thunks.ts (hunk 3 above)
Comment (Sm00g15) [reply src/reducers/app/thunks.ts]: not sure where this code came from, investigating now.

File: src/containers/CustomTable/index.tsx (hunk 2 above)
Comment (Sm00g15) [reply src/containers/CustomTable/index.tsx]: Removed and pushed.

File: src/reducers/app/thunks.ts (hunk 3 above)
Comment (Sm00g15) [reply src/reducers/app/thunks.ts]: We spoke about using a filter function yesterday, but I haven't found a way with filter to return them in the particular order that we want, I pushed up the changes using the reduce function but if you have a better idea using filter I'm all ears.

File: src/reducers/app/thunks.ts (hunk 4 above)
Comment (Sm00g15) [reply src/reducers/app/thunks.ts]: We spoke about using a filter function yesterday, but I haven't found a way with filter to return them in the particular order that we want, I pushed up the changes using the reduce function but if you have a better idea using filter I'm all ears.

**Pull request merged**